### PR TITLE
Feature/sprint crud

### PR DESCRIPTION
**Pull request by maksimowiczm**

### CODE REVIEW

File: src/core/Codend.Application/Sprints/Commands/UpdateSprint/UpdateSprintCommand.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        if (sprint is null)
+        {
+            return DomainNotFound.Fail<Sprint>();
+        }
+
+        Result<SprintPeriod> periodResult = Result.Ok();
+        if (request.StartDate is not null && request.EndDate is not null)
+        {
+            periodResult = sprint.EditPeriod(
+                request.StartDate.Value.ToUniversalTime(),
```
Comment (pzajaczkowski) [inline src/core/Codend.Application/Sprints/Commands/UpdateSprint/UpdateSprintCommand.cs:61]: We have json converter that will convert datetime string into UTC datetime. So we don't need it here. We have to be sure to send proper utc (ISO 8601 standard) string to backend.

File: src/core/Codend.Application/Sprints/Commands/UpdateSprint/UpdateSprintValidator.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        When(x => x.Name is not null, () =>
+        {
+            RuleFor(x => x.Name)
+                .NotEmpty()
+                .WithError(new PropertyNullOrEmpty(nameof(UpdateSprintCommand.Name)))
+                .MaximumLength(SprintName.MaxLength)
+                .WithError(new StringPropertyTooLong(nameof(UpdateSprintCommand.Name), StoryName.MaxLength));
+        });
+        
+        RuleFor(x => x.StartDate)
```
Comment (pzajaczkowski) [inline src/core/Codend.Application/Sprints/Commands/UpdateSprint/UpdateSprintValidator.cs:31]: I think there should be rule checking if enddate is not after startdate.
Edit: or we can leave it as it is if you think period value object own validation is enough and nothing will be infected after.

**Review by pzajaczkowski — APPROVED** [review]

File: src/core/Codend.Application/Sprints/Commands/UpdateSprint/UpdateSprintCommand.cs (hunk 1 above)
Comment (maksimowiczm) [reply src/core/Codend.Application/Sprints/Commands/UpdateSprint/UpdateSprintCommand.cs:61]: So you telling me that we assume that we get valid data from the client. 🤔

Comment (maksimowiczm) [reply src/core/Codend.Application/Sprints/Commands/UpdateSprint/UpdateSprintCommand.cs:61]: DateTime is stored with timezone and Postgres can't store timezones. If we don't call this method database will just refuse to store the date.

Comment (pzajaczkowski) [reply src/core/Codend.Application/Sprints/Commands/UpdateSprint/UpdateSprintCommand.cs:61]: Yes this ISO convention assures that the time is sent in utc standard. There are some libraries in js but normal Date() should work as well. Idk what are you talking about postgres not having timezones (timestampz) 

image

Nevertheless we should be aware of possible time problems.

Comment (maksimowiczm) [reply src/core/Codend.Application/Sprints/Commands/UpdateSprint/UpdateSprintCommand.cs:61]: dobra nie zesraj sie

Comment (maksimowiczm) [reply src/core/Codend.Application/Sprints/Commands/UpdateSprint/UpdateSprintCommand.cs:61]: 💀️

Comment (maksimowiczm) [reply src/core/Codend.Application/Sprints/Commands/UpdateSprint/UpdateSprintCommand.cs:61]: Postgres timestamp with time zone support only UTC.

**Pull request merged**